realm/realm-library/src/main/java/io/realm/RealmFieldType.java

```diff
@@ -76,22 +76,22 @@ public int getNativeValue() {
      */
     public boolean isValid(Object obj) {
         switch (nativeValue) {
-            case 0: return (obj instanceof Long || obj instanceof Integer || obj instanceof Short || obj instanceof Byte);
-            case 1: return (obj instanceof Boolean);
-            case 2: return (obj instanceof String);
-            case 4: return (obj instanceof byte[] || obj instanceof ByteBuffer);
-            case 5: return (obj == null || obj instanceof Object[][]);
-            case 6: return (obj instanceof Mixed ||
+            case 0: return obj instanceof Long || obj instanceof Integer || obj instanceof Short || obj instanceof Byte;
```

Comment:
If you argued that `(obj instanceof Long) || (obj instanceof Integer) || (obj instanceof Short) || (obj instanceof Byte)` is easier to read, I might agree. But a single parenthesis doesn't.